session.go

```diff
@@ -607,15 +608,17 @@ func createSessionWithDomainFunc(store kv.Storage) func(*domain.Domain) (pools.R
 
 func drainRecordSet(ctx context.Context, rs ast.RecordSet) ([]types.Row, error) {
 	var rows []types.Row
-	for {
-		row, err := rs.Next(ctx)
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
-		if row == nil {
-			break
-		}
-		rows = append(rows, row)
+	chk := rs.NewChunk()
```

Comment:
line 611~621 should be warped with a `for` loop to insure we have got all the results of `rs`